### ISSUE TITLE
Reproduce-check failures are hard to diagnose: the produced tensor is discarded, and the rtol/atol extremes reported are meaningless for integer label outputs

### ISSUE THREAD

**Issue by Cyerme** [bioimage-io/core-bioimage-io-python#502]
## Context

Packaging a `pytorch_state_dict` model whose single output is an **integer
instance-label map**. Local `bioimageio test` is bit-exact (`max difference
0.00e+00`). The same package on a remote BioEngine worker (different GPU)
disagreed on 437 of 1 016 064 elements (430.1 ppm) — a one-pixel wobble on some
instance boundaries, caused by fp16 autocast accumulation order. Two properties
of the failure path made that far harder to establish than it needed to be.

## 1. The produced tensor is not saved, so you cannot see what the run produced

`_test_recreate_test_outputs` writes the produced tensors only when
`working_dir is not None and verbose`, and `verbose` is itself derived from
`working_dir`:

```python
verbose = working_dir is not None        # _resource_tests.py:284
...
verbose=working_dir is not None          # _resource_tests.py:765
```

When it does write, it appends `"Saved (intermediate) outputs to
{output_paths}."` to the message. That sentence was absent from the worker's
report, which is how we know the worker's tensor is simply gone.

The result is that a reproduce failure on someone else's machine gives you a
scalar summary and nothing to inspect. We could only characterise the deviation
by *reconstructing* it locally — we eventually reproduced it at 467 px / 459.6 ppm
by setting
`torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = False`,
which is a lot of work to answer "which pixels moved, and did the instance count
change?"

**Suggestion:** always persist the produced tensor (or at least a diff mask /
index list of mismatched elements) into the report's working directory, or add a
report field carrying it, independent of `verbose`. For remote runners, even a
compressed boolean mismatch mask would be enough to tell a boundary wobble from a
structural change.

## 2. The reported extremes are meaningless for discrete-valued outputs

`evaluate_mismatched_elements` casts both arrays to `float32` and always appends:

```
Max relative difference: 2.30e+07 (|a - e|/|e + 1e-6|)
Max absolute difference: 2.50e+01
```

On an instance-label map those numbers carry no information. A boundary pixel
that flips from background `0` to instance `23` yields a relative difference of
2.3e7 purely because the denominator is `0 + 1e-6`. A reader — or a curator
skimming a report — sees "max relative difference 2.3e+07" and reasonably
concludes the model is broken, when the actual finding is "437 boundary pixels
of a million moved by one".

The same message is emitted even when the check **passes** on the ppm criterion
alone, which is the confusing case: a passing check whose printed diagnostics
look catastrophic.

**Suggestion:** when the expected tensor's dtype is integer (or when the check
passed on `mismatched_elements_per_million` while rtol/atol were not the binding
constraint), suppress or qualify the rtol/atol extremes and report the
discrete-appropriate quantities instead — number and fraction of mismatched
elements, and ideally the number of distinct labels in each array. Something like:

```
9 of 1016064 elements (8.9 ppm) disagree with expected values.
Output dtype is integer; relative/absolute tolerances are not applicable.
Distinct labels: actual 41, expected 41.
```

## 3. Minor: the environment comparison in the dynamic-test log reports the wrong runtime

The trailing "Conda Environment Comparison" block in our test log listed
Python 3.12.1 and torch 2.10.0, neither of which was the environment the test
actually ran in (Python 3.11.16 / torch 2.8.0+cu128). If that section is meant
to describe the runtime that produced the report, it is misleading; if it
describes something else (a reference environment), the heading should say so.

## Environment

- bioimageio.core: 0.11.0
- bioimageio.spec: 0.5.12.0
- Python: 3.11.16
- torch: 2.8.0+cu128 (CUDA 12.8)
- Model: `pytorch_state_dict`, format_version 0.5.12, one integer-labelled
  output tensor of 1 016 064 elements

**Comment by FynnBe** [comment]
Thanks @Cyerme for the issue!
1. Not sure if I missed something, but why did `verbose=True` not solve this issue?
2. Great suggestions to improve the message for integer outputs! I was a bit hesitant in the past to go that route as we do not have the semantic labels to identify 'label' outputs. I do think the proxy of integer dtype is a safe (enough) assumption though.
3. Interesting. I will need to investigate this further...

**Comment by FynnBe** [comment]
Re 3. the conda environment comparison (and listing) assumes the running Python executable is the one from the active conda environment. Maybe this didn't hold true in your case?
